weasyl/searchtag.py

```diff
@@ -227,6 +240,10 @@ def associate(userid, tags, submitid=None, charid=None, journalid=None):
     added = entered_tagids - existing_tagids
     removed = existing_tagids - entered_tagids
 
+    # enforce the limit on artist preference tags
+    if artistid and (len(added) - len(removed) + len(existing)) > 50:
```

Comment:
50 should be moved to a constant near the top of the file (e.g. `MAX_PREFERRED_TAGS = 50` as opposed to hidden in the code.